File: app/src/test/java/com/msaggik/algorithmsdatastructures/java/RecursiveAlgorithmTest.java
package com.msaggik.algorithmsdatastructures.java;

import org.junit.Assert;
import org.junit.Test;

public class RecursiveAlgorithmTest {

    @Test(timeout = 100)
    public void testCalculationFactorialOne() {
        long result = RecursiveAlgorithm.calculationFactorial(10L);
        Assert.assertEquals(result, 3_628_800L);
    }

    @Test(timeout = 100)
    public void testCalculationFactorialTwo() {
        long result = RecursiveAlgorithm.calculationFactorial(14L);
        Assert.assertEquals(result, 87_178_291_200L);
    }

    @Test(timeout = 100)
    public void testCalculationFibonacciOne() {
        long result = RecursiveAlgorithm.calculationValueFibonacci(10L);
        Assert.assertEquals(result, 55);
    }

    @Test(timeout = 100)
    public void testCalculationFibonacciTwo() {
        long result = RecursiveAlgorithm.calculationValueFibonacci(14L);
        Assert.assertEquals(result, 377);
    }
}

File: app/src/test/java/com/msaggik/algorithmsdatastructures/kotlin/RecursiveAlgorithmTest.kt
package com.msaggik.algorithmsdatastructures.kotlin

import com.msaggik.algorithmsdatastructures.kotlin.RecursiveAlgorithm
import org.junit.Assert
import org.junit.Test

class RecursiveAlgorithmTest {
    @Test(timeout = 100)
    fun testCalculationFactorialOne() {
        val result = RecursiveAlgorithm.calculationFactorial(10L)
        Assert.assertEquals(result, 3628800L)
    }

    @Test(timeout = 100)
    fun testCalculationFactorialTwo() {
        val result = RecursiveAlgorithm.calculationFactorial(14L)
        Assert.assertEquals(result, 87178291200L)
    }

    @Test(timeout = 100)
    fun testCalculationFibonacciOne() {
        val result = RecursiveAlgorithm.calculationValueFibonacci(10L)
        Assert.assertEquals(result, 55)
    }

    @Test(timeout = 100)
    fun testCalculationFibonacciTwo() {
        val result = RecursiveAlgorithm.calculationValueFibonacci(14L)
        Assert.assertEquals(result, 377)
    }
}
